input/full.go

```diff
@@ -29,21 +29,27 @@ import (
 func CollectFull(ctx context.Context, server *state.Server, connection *sql.DB, opts state.CollectionOpts, logger *util.Logger) (ps state.PersistedState, ts state.TransientState, err error) {
 	ps.CollectedAt = time.Now()
 
+	// These were previously run in separate goroutines, but because buffer cache checks
+	// during Aurora Serverless scaling can cause segfaults, we now run them in the same
+	// goroutine so that we can recognize Serverless nodes from the API (there does not
+	// seem to be a way to check whether Aurora is Serverless in SQL).
```

Comment:
I'd reword this a bit, I don't think the "previously ..." style makes sense for comments (I'd usually only use that in commit messages), but I'd instead describe why the code works the way it does. Maybe:

```suggestion
    // Buffer cache statistics and system information are technically separate, but there is
    // an edge case we care about: We intentionally turn off use of pg_buffercache data
    // when Amazon Aurora Serverless is in use, because it currently has a bug that causes
    // a server crash during a scaling event, if pg_buffercache is queried concurrently.
```

Reply:
```suggestion
    // Buffer cache statistics and system information are separate, but querying pg_buffercache during
    // an Amazon Aurora Serverless scaling event can cause a server crash (which seems to be due to
    // a Serverless bug), so we first get system information and skip the buffer cache if necessary.
```
Yeah, good point. Reworded a bit to note that the system information check is to determine if we should skip the buffer cache scan (and for conciseness).